src/psyclone/tests/psyir/transformations/intrinsics/dotproduct2code_trans_test.py

```diff
@@ -537,3 +551,31 @@ def test_apply_multi_different_line(tmpdir, fortran_reader, fortran_writer):
         "  enddo\n"
         "  result = result + c * res_dot_product_1\n")
     check_trans(code, expected, fortran_reader, fortran_writer, tmpdir)
+
+
+def test_apply_explicit_range(fortran_reader, fortran_writer, tmpdir):
```

Comment:
This test fails for me when compilation is enabled:

    ----------------------------- Captured stderr call -----------------------------
    Compiling: gfortran generated-8788527625436.f90 -c generated-8788527625436.f90:3:32:

        3 |   real, dimension(:) :: basis_w1
          |                                1
    Error: Array ‘basis_w1’ at (1) cannot have a deferred shape
    
    =========================== short test summary info ============================
    FAILED psyir/transformations/intrinsics/dotproduct2code_trans_test.py::test_apply_explicit_range

Reply:
Oops. Fixed.